### PR TITLE
Taproot & MuSig2 pre-lims

### PR DESCRIPTION
**Pull request by ellemouton**
Feedback welcome :) I suggest viewing in the "Files Changed" mode instead of by commit for this. 

see the post with diagrams here: https://ellemouton.com/drafts/taproot-prelims

### CODE REVIEW

File: content/drafts/taproot-prelims.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+can be found in [BIP86][bip86-tweak]. 
+
+# MuSig2
+
+With the Taproot softfork, bitcoin nodes now have an understanding of BIP340 
+signatures (or Schnorr signatures). The beauty of these signatures is their 
+linearity: the owner of public key `P_1` can create a partial signature, 
+`sig_1` for the message `msg` and the owner of public key `P_2` can create 
+`sig_2` for the same message. The two parties will then be able to combine their 
+signatures such that `sig` where `sig = `sig_1 + sig_2` is a valid signature for
```
Comment (WalterSmuts) [inline content/drafts/taproot-prelims.md:202]: Missing a backtick here

File: content/drafts/taproot-prelims.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+n-of-n multisig script and then needing to pay for the blockchain space to store 
+each of the `n` signatures, only one signature will be needed and no long script 
+will be required at all. Instead, only a single public key (which is actually an 
+_aggregate_ public key) needs to appear on-chain.
+
+The tricky part here is everything that needs to happen _off-chain_ during the 
+setup of this aggregate public key as well as for the creation of the final 
+signature. MuSig2 is the protocol that defines how this should be done. The 
+various steps have been carefully thought through in order to keep the process 
+trust-less and to protect parties from attacks such as key cancellation.
```
Comment (WalterSmuts) [inline content/drafts/taproot-prelims.md:215]: Link?

File: content/drafts/taproot-prelims.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+## MuSig2 vs n-of-n Multisig
+
+An important thing to keep in mind is that with n-of-n multisig outputs, parties
+can generate their signatures completely independently of the other parties. As 
+long as they have the message to be signed along with their private key, they 
+can create a signature. This signature can then safely be distributed to the 
+other parties and eventually the transaction witness will have all `n` 
+signatures. In other words: no interaction is required between parties at 
+signing time. With n-of-n _MuSig2_ this is not the case since there is one
```
Comment (WalterSmuts) [inline content/drafts/taproot-prelims.md:234]: `With n-of-n _MuSig2_` -> `With n-of-n _Multisig_`

File: content/drafts/taproot-prelims.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+![](/taprootPrelims/musig-6.png#center)
+
+It is important to become familiar with the above MuSig2 flow because it will be
+used very often in the next few articles. In Taproot channels the funding output 
+of a channel will be a MuSig2 aggregate public key. This means that every 
+commitment transaction created that spends from the funding output will need to 
+go through this signing flow. Since channel states in Lightning are asymmetric, 
+this also means that this flow will need to happen twice per state update: once 
+to sign the local commitment transaction and one to sign the remote one. But
```
Comment (WalterSmuts) [inline content/drafts/taproot-prelims.md:294]: one -> once

**Review by WalterSmuts — COMMENTED** [review]
NICE!

**Review by ellemouton — COMMENTED** [review]
Dankie @WalterSmuts! Updated :)

File: content/drafts/taproot-prelims.md [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+put in what is called the “control block”. The first thing in the control block 
+is the internal key, `P`. It also contains the Merkle proof that allows the 
+verifier to compute the Merkle tree root. The witness already includes 
+`Script B` itself (it was required for step 1), so the verifier can compute 
+`hB` (see the diagram above showing the Merkle tree construction) themselves, 
+and so we just need to provide `hA` and `hC`. The validator will use these 
+hashes to calculate the `script_root` and then hash this along with the 
+internal key, `P`, in order to arrive at the tweak, `t`. The validator can then 
+compute the corresponding tweak point, `T`, add that to the internal key, `P`, 
+to get the output key `Q`. The final thing that the control block must include
```
Comment (GeorgeTsagk) [inline content/drafts/taproot-prelims.md:167]:
```suggestion
to get the output key, `Q`. The final thing that the control block must include 
```

File: content/drafts/taproot-prelims.md [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+done. The various steps have been carefully thought through in order to keep 
+the process trust-less and to protect parties from attacks such as key 
+cancellation.
+
+[BIP327][bip327] defines the MuSig2 protocol along with a bunch of algorithms 
+that should be used for the various steps of the process. Since the aim of this 
+article is to provide all the building blocks required for understanding 
+Lightning Taproot channels, I will only talk about MuSig2 at an API level using 
+the defined algorithms and will focus more on how it will be used in Lightning. 
+If you would like to dig into it more you, can check out the BIP itself. I have
```
Comment (GeorgeTsagk) [inline content/drafts/taproot-prelims.md:223]:
```suggestion
If you would like to dig into it more you can check out the BIP itself. I have 
```

**Review by GeorgeTsagk — APPROVED** [review]
lgtm! great content

I did find some minory-minor nits

File: content/drafts/taproot-prelims.md (hunk 6 above)
Comment (ellemouton) [reply content/drafts/taproot-prelims.md:223]: hehehe - don't know how this one got there 😂

**Comment by ellemouton** [comment]
Thanks @GeorgeTsagk! Updated 👍

**Comment by Liongrass** [comment]
Reads very well!

**Comment by ellemouton** [comment]
taking this live now :) thanks for the input everyone! <3

**Pull request closed without merging**